UnitTestFrameworkPkg/UnitTestFrameworkPkgHost.dsc.inc

```diff
@@ -69,22 +102,40 @@
   MSFT:*_VS2022_X64_DLINK_FLAGS     = /LIBPATH:"%VCToolsInstallDir%lib\x64" /LIBPATH:"%UniversalCRTSdkDir%lib\%UCRTVersion%\ucrt\x64" /LIBPATH:"%WindowsSdkDir%lib\%WindowsSDKLibVersion%\um\x64"
 
   #
-  # GCC
+  # Set GCC HOST_APPLICATION options
   #
-  GCC:*_*_*_CC_FLAGS       =  -Wno-write-strings
-  GCC:*_*_IA32_DLINK_FLAGS == -o $(BIN_DIR)/$(MODULE_NAME_GUID) -m32 -no-pie
-  GCC:*_*_X64_DLINK_FLAGS  == -o $(BIN_DIR)/$(MODULE_NAME_GUID) -m64 -no-pie
+  GCC:*_*_*_CC_FLAGS        =  -Wno-write-strings
+  GCC:*_*_IA32_DLINK_FLAGS  == -o $(BIN_DIR)/$(MODULE_NAME_GUID) -z noexecstack -m32 -no-pie
+  GCC:*_*_X64_DLINK_FLAGS   == -o $(BIN_DIR)/$(MODULE_NAME_GUID) -z noexecstack -m64 -no-pie
+!if $(UNIT_TESTING_ADDRESS_SANITIZER_ENABLE)
+  #
+  # Enable Address Sanitizer linker options
+  #
+  GCC:*_*_*_DLINK_FLAGS    = -fsanitize=address
+!endif
+!if $(UNIT_TESTING_CODE_COVERAGE_ENABLE)
+  #
+  # Enable code coverage linker options
+  #
+  GCC:*_GCC_*_DLINK_FLAGS        = --coverage
+  GCC:*_GCC5_*_DLINK_FLAGS       = --coverage
+  GCC:*_GCCNOLTO_*_DLINK_FLAGS   = --coverage
+  GCC:*_CLANGDWARF_*_DLINK_FLAGS = -fprofile-instr-generate -fcoverage-mapping
+!endif
   #
   # Surround our static libraries with whole-archive, so constructor-based test registration works properly.
   # Note that we need to --no-whole-archive before linking system libraries.
   #
   GCC:*_*_*_DLINK_FLAGS    = -Wl,--whole-archive
-  GCC:*_*_*_DLINK2_FLAGS   == -Wl,--no-whole-archive -lgcov -lpthread -lstdc++ -lm
-!if $(UNIT_TESTING_ADDRESS_SANITIZER_ENABLE)
+  GCC:*_*_*_DLINK2_FLAGS   == -Wl,--no-whole-archive -lpthread -lstdc++ -lm
+
   #
-  # Enable Address Sanitizer for GCC for HOST_APPLICATION
+  # Set CLANGPDB HOST_APPLICATION options
   #
-  GCC:*_*_*_DLINK_FLAGS    = -fsanitize=address
+  CLANGPDB:*_*_*_DLINK_FLAGS   == /nologo /SUBSYSTEM:CONSOLE /DEBUG /out:"$(BIN_DIR)\$(MODULE_NAME_GUID).exe" /pdb:"$(BIN_DIR)\$(MODULE_NAME_GUID).pdb" /WHOLEARCHIVE
+  CLANGPDB:*_*_X64_DLINK_FLAGS  = /LIBPATH:"%VCToolsInstallDir%lib\x64" /LIBPATH:"%UniversalCRTSdkDir%lib\%UCRTVersion%\ucrt\x64" /LIBPATH:"%WindowsSdkDir%lib\%WindowsSDKLibVersion%\um\x64"
+!if $(UNIT_TESTING_ADDRESS_SANITIZER_ENABLE)
+  CLANGPDB:*_*_X64_DLINK_FLAGS  = clang_rt.profile-x86_64.lib clang_rt.asan_dynamic-x86_64.lib clang_rt.asan_dynamic_runtime_thunk-x86_64.lib
 !endif

```

Comment:
Were `IA32` library intentionally not set here? IA32 still seems supported in the rest of the build options.

Reply:
The libs listed in this line are only or X64 so that line is correct.

I do not know the correct settings for IA32 CLANGPDB host unit test builds yet, and stuart build of unit tests only allow X64 due to other IA32 related issues.  We can leave IA32 CLANGPDB host unit test build support as an open issue for now.